pandas/core/indexes/numeric.py

```diff
@@ -229,7 +249,10 @@ def astype(self, dtype, copy=True):
 
     @doc(Index._should_fallback_to_positional)
     def _should_fallback_to_positional(self) -> bool:
-        return False
+        if self.inferred_type == "floating":
```

Comment:
simpler `self.dtype.kind == "f"`?

Reply:
`if is_float_dtype(self.dtype)` is IMO even more idiomatic. Changed to that.